core/src/main/java/de/eldoria/jacksonbukkit/entities/RecipeChoiceWrapper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *     SPDX-License-Identifier: MIT
+ *
+ *     Copyright (C) EldoriaRPG Team and Contributor
+ */
+package de.eldoria.jacksonbukkit.entities;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.bukkit.inventory.RecipeChoice;
+
+/**
+ * Class for wrapping a {@link RecipeChoice}.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "@clazz")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = ExactChoiceWrapper.class, name = "ExactChoiceWrapper"),
+        @JsonSubTypes.Type(value = MaterialChoiceWrapper.class, name = "MaterialChoiceWrapper")
```

Comment:
```suggestion
        @JsonSubTypes.Type(value = ExactChoiceWrapper.class, name = "ExactChoice"),
        @JsonSubTypes.Type(value = MaterialChoiceWrapper.class, name = "MaterialChoice")
```

Dunno if we should use the original names for that. ...Wrapper can be seens as implementation detail.